### ISSUE TITLE
Optional params filled in, leading to error

### ISSUE THREAD

**Issue by jtojnar** [slince/di#5]
When I try to instantiate [`Monolog\Logger`](https://github.com/Seldaek/monolog/blob/2.8.0/src/Monolog/Logger.php#L186), the  process fails on trying to fill in an optional argument:

```
Fatal error: Uncaught Slince\Di\Exception\DependencyInjectionException: Unresolvable dependency resolving "timezone" in class "DateTimeZone" in Resolver.php:173
Stack trace:
#0 Resolver.php(91): Slince\Di\Resolver->resolveDependencies(Array, Array)
#1 Resolver.php(46): Slince\Di\Resolver->createFromClass(Object(Slince\Di\Definition))
#2 Container.php(256): Slince\Di\Resolver->resolve(Object(Slince\Di\Definition))
#3 Container.php(220): Slince\Di\Container->resolveInstance('DateTimeZone')
#4 Resolver.php(164): Slince\Di\Container->get('DateTimeZone')
#5 Resolver.php(91): Slince\Di\Resolver->resolveDependencies(Array, Array)
#6 Resolver.php(46): Slince\Di\Resolver->createFromClass(Object(Slince\Di\Definition))
#7 Container.php(256): Slince\Di\Resolver->resolve(Object(Slince\Di\Definition))
#8 Container.php(220): Slince\Di\Container->resolveInstance('Logger')
#9 test.php(15): Slince\Di\Container->get('Logger')
#10 {main}
  thrown in Resolver.php on line 173
```

I have managed to reproduce it with the following simple file:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$container = new Slince\Di\Container();
$container->setDefaults(['shared' => false]);
$container
    ->register(Monolog\Logger::class)
    ->addArgument('selfoss')
    ->setShared(true);


$log = $container->get(Monolog\Logger::class);
```

**Comment by slince** [comment]
thanks your feedback, fix in v3.2.1

**Comment by jtojnar** [comment]
Thanks, seems to work.

Unfortunately, we are still targetting PHP 7.2 since that is what [Ubuntu LTS supports](https://www.github.com/FreshRSS/FreshRSS/discussions/3321#discussioncomment-835704) so we will need to figure out something else.
